test/arborist/build-ideal-tree.js

```diff
@@ -3110,3 +3163,522 @@ t.test('competing peerSets resolve in both root and workspace', t => {
+      }),
+    })
+
+    await t.rejects(buildIdeal(path), { code: 'EOVERRIDE' }, 'throws EOVERRIDE')
+  })
+
+  t.test('overrides a nested dependency', async (t) => {
+    generateNocks(t, {
+      foo: {
+        versions: ['1.0.0', '1.0.1', '2.0.0'],
+        dependencies: ['bar'],
+      },
+      bar: {
+        versions: ['1.0.0', '1.0.1', '2.0.0'],
+      },
+    })
+
+    const path = t.testdir({
+      'package.json': JSON.stringify({
+        name: 'root',
+        dependencies: {
+          foo: '1.0.1',
+        },
+        overrides: {
+          bar: '2.0.0',
+        },
+      }),
+    })
+
+    const tree = await buildIdeal(path)
+
+    const fooEdge = tree.edgesOut.get('foo')
+    t.equal(fooEdge.valid, true)
+
+    const fooBarEdge = fooEdge.to.edgesOut.get('bar')
+    t.equal(fooBarEdge.valid, true)
+    t.equal(fooBarEdge.to.version, '2.0.0')
+  })
+
+  t.test('overrides a nested dependency with a more specific override', async (t) => {
+    generateNocks(t, {
+      foo: {
+        versions: ['1.0.0', '1.0.1', '2.0.0'],
+        dependencies: ['bar'],
+      },
+      bar: {
+        versions: ['1.0.0', '1.0.1', '2.0.0'],
+      },
+    })
+
+    const path = t.testdir({
+      'package.json': JSON.stringify({
+        name: 'root',
+        dependencies: {
+          foo: '1.0.1',
+          bar: '2.0.0',
+        },
+        overrides: {
+          foo: {
+            bar: '2.0.0',
+          },
+        },
+      }),
+    })
+
+    const tree = await buildIdeal(path)
+
+    const fooEdge = tree.edgesOut.get('foo')
+    t.equal(fooEdge.valid, true)
+
+    const fooBarEdge = fooEdge.to.edgesOut.get('bar')
+    t.equal(fooBarEdge.valid, true)
+    t.equal(fooBarEdge.to.version, '2.0.0')
+
+    const barEdge = tree.edgesOut.get('bar')
+    t.equal(barEdge.valid, true)
+    t.equal(barEdge.to.version, '2.0.0')
+  })
+
+  t.test('does not override a nested dependency when parent spec does not match', async (t) => {
+    generateNocks(t, {
+      foo: {
+        versions: ['1.0.0', '1.0.1', '2.0.0'],
+        dependencies: ['bar'],
+      },
+      bar: {
+        versions: ['1.0.0', '1.0.1', '2.0.0'],
+      },
+    })
+
+    const path = t.testdir({
+      'package.json': JSON.stringify({
+        name: 'root',
+        dependencies: {
+          foo: '1.0.1',
+          bar: '2.0.0',
+        },
+        overrides: {
+          'foo@2': {
+            bar: '2.0.0',
+          },
+        },
+      }),
+    })
+
+    const tree = await buildIdeal(path)
+
+    const fooEdge = tree.edgesOut.get('foo')
+    t.equal(fooEdge.valid, true)
+
+    const fooBarEdge = fooEdge.to.edgesOut.get('bar')
+    t.equal(fooBarEdge.valid, true)
+    t.equal(fooBarEdge.to.version, '1.0.1')
+
+    const barEdge = tree.edgesOut.get('bar')
+    t.equal(barEdge.valid, true)
+    t.equal(barEdge.to.version, '2.0.0')
+  })
+
+  t.test('overrides a nested dependency that also exists as a direct dependency', async (t) => {
+    generateNocks(t, {
```

Comment:
You use this same Nock fixture quite a bit. It could be done once and referenced.

Reply:
i liked keeping the calls to generateNocks inline in the test, it's handy to have the context of what a package's dependency tree looks like right in the test that exercises it. part of the struggles with the current testing that i experienced was having to go read fixtures to find out what the dep tree was even supposed to be.

it's a first pass at introducing nock here though, and i'd wager there's lots of improvements to be made